ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -171,9 +176,12 @@ public void run() {
               StringUtils.stringifyException(t));
         }
         finally {
-          if(handle != null) {
+          if (handle != null) {
             handle.releaseLocks();
           }
+          if (metricsEnabled) {
```

Comment:
Won't this fail, if the acquireLock times out and the perflogger was not started? The PerfLogger in ql has a method for checking this startTimeHasMethod, maybe it worth to copy that and use it here

Reply:
checked: it won't fail and won't report metric in this case